src/mixins/date-mixin.ts

```diff
@@ -171,6 +171,27 @@ export default class DateMixin extends Mixins(CommonMixin) {
     return `${dateStr} at ${timeStr}`
   }
 
+  /**
+   * Converts a date object to an ISO datetime string format that the API accepts (YYYY-MM-DDThh:mm:ss.ssssss+00:00)
+   # @example "2021-01-01 23:59:59 PST" -> "2021-01-02T07:59:59.000000+00:00"
+   */
+  dateToApiIsoDateTimeString (date: Date): string {
+    if (!date) return null // safety check
+
+    // convert date object to ISO format that API accepts
+    // eg, 2021-03-25T10:02:00.000000+00:00
+    const dateStr = date.getUTCFullYear() +
+      '-' + this.pad(date.getUTCMonth() + 1) +
+      '-' + this.pad(date.getUTCDate()) +
+      'T' + this.pad(date.getUTCHours()) +
+      ':' + this.pad(date.getUTCMinutes()) +
+      ':' + this.pad(date.getUTCSeconds()) +
+      '.' + (date.getUTCMilliseconds() / 1000).toFixed(6).slice(2, 8) +
+      '+00:00'
```

Comment:
In my opinion, this is not the correct solution.

Admittedly, both the output of `date.toISOString()` (eg, 2021-01-02T07:59:59.000Z) and the Python format (eg, 2021-01-02T07:59:59.000Z) are ISO 8601, but apparently aren't compatible.

Given the examples above, it should be trivial to convert from one to the other. I would simply knock off the "Z" and append the offset string. See apiToDateAndTimeString() above and reverse the process.

Reply:
so if i re-work this function(`dateToApiIsoDateTimeString`) to use `date.toISOString()` as a starting point, I would still be missing part of the milliseconds that the API expects.  What do we want to do about this?  Should we replace the Z with 000+00:00 ?  Or should I just grab the milliseconds with what I was originally using?  i.e. `(date.getUTCMilliseconds() / 1000).toFixed(6).slice(2, 8)`

Reply:
As it turns out, `toISOString()` gives you the milliseconds, eg
```
> new Date().toISOString()
> "2021-03-22T20:20:56.052Z"
```

How inflexible is the Python ISO date parser? Do you have specify it all exactly? If so then, yes, you'll have to right-pad the milliseconds (and add the tz offset).

This is what I would do (but feel free to do something nicer!):
```
const dateStr = date.toISOString().slice(1,23) + '000+00:00'
```

Ref: https://www.w3schools.com/jsref/jsref_slice_array.asp

Reply:
^^ and add a code comment describing what you're doing ;)

Reply:
For some reason I thought I needed the additional milliseconds.  I just tested it and the api did accept a format of `2021-03-26T13:04:00.000Z`.  I might have been just thinking we might need the millisecond precision part which is highly unlikely if I think about it.

So I'll just do as per your suggestion:

```
const dateStr = date.toISOString().slice(1,23) + '+00:00'
```